From 3d013e0f3e0149ae6616e273ac09dd1da61f59b9 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Thu, 14 Nov 2024 11:05:46 -0700
Subject: [PATCH] fix: Allow for environment override of CMAKE_CXX_FLAGS

* For non-x86 platforms '-march=nocona -mtune=haswell' are not necessarily
  supported, so allow for CMAKE_CXX_FLAGS to use its default behavior of taking
  the value of CXXFLAGS if it exists in the environment.
   - c.f. https://cmake.org/cmake/help/v3.31/envvar/CXXFLAGS.html
---
 CMakeLists.txt | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 412b98c..e790d05 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,16 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(VERSION "\"2.0.9\"")
-set(CMAKE_CXX_FLAGS "-Wall -Wextra -march=nocona -mtune=haswell -fvisibility-inlines-hidden -fmessage-length=0 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -fext-numeric-literals")
+# Set default CXXFLAGS but allow for environment override
+if (NOT CMAKE_CXX_FLAGS)
+  set(CMAKE_CXX_FLAGS "-Wall -Wextra -march=nocona -mtune=haswell -fvisibility-inlines-hidden -fmessage-length=0 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -fext-numeric-literals")
+else()
+  # Ensure -fext-numeric-literals is in CMAKE_CXX_FLAGS
+  string(FIND "${CMAKE_CXX_FLAGS}" "-fext-numeric-literals" _found_ext_numeric_literals)
+  if (_found_ext_numeric_literals EQUAL -1)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fext-numeric-literals" CACHE STRING "Update environment CXXFLAGS" FORCE)
+  endif()
+endif()
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fsanitize=address -fPIC ${CMAKE_CXX_FLAGS}" CACHE STRING "debug compile flags" FORCE)
 set(CMAKE_EXE_LINKER_FLAGS_DEBUG "-fsanitize=address" CACHE STRING "debug linker flags" FORCE)
 set(prefix ${CMAKE_INSTALL_PREFIX})
-- 
2.47.0

